File: libraries/clib/native/Math.c
/*
 * java.lang.Math.c
 *
 * Copyright (c) 1996, 1997
 *	Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "lib-license.terms" for information on usage and redistribution 
 * of this file. 
 */

#include "config.h"
#include "config-std.h"
#include <native.h>
#include <math.h>
#include "config-math.h"
#include "Math.h"

double
java_lang_Math_sin(double v)
{
	return (sin(v));
}

double
java_lang_Math_cos(double v)
{
	return (cos(v));
}

double
java_lang_Math_tan(double v)
{
	return (tan(v));
}

double
java_lang_Math_asin(double v)
{
	return (asin(v));
}

double
java_lang_Math_acos(double v)
{
	return (acos(v));
}

double
java_lang_Math_atan(double v)
{
	return (atan(v));
}

double
java_lang_Math_exp(double v)
{
	return (exp(v));
}

double
java_lang_Math_log(double v)
{
	return (log(v));
}

double
java_lang_Math_sqrt(double v)
{
	return (sqrt(v));
}

double
java_lang_Math_IEEEremainder(double v1, double v2)
{
	return (IEEERemainder(v1, v2));
}

double
java_lang_Math_ceil(double v)
{
	return (ceil(v));
}

double
java_lang_Math_floor(double v)
{
	return (floor(v));
}

double
java_lang_Math_rint(double v)
{
	/* As per Java Class Libraries page 927 */
	return (floor(v+0.5));
}

double
java_lang_Math_atan2(double v1, double v2)
{
	return (atan2(v1, v2));
}

double
java_lang_Math_pow(double v1, double v2)
{
	return (pow(v1, v2));
}

File: libraries/javalib/java/lang/Math.java
package java.lang;

import java.util.Random;

/*
 * Java core library component.
 *
 * Copyright (c) 1997, 1998
 *      Transvirtual Technologies, Inc.  All rights reserved.
 *
 * See the file "license.terms" for information on usage and redistribution
 * of this file.
 */
final public class Math
{
	final public static double E = 2.7182818284590452354;
	final public static double PI = 3.14159265358979323846;
	public static Random randomGenerator = new Random();

native public static double IEEEremainder(double f1, double f2);

public static double abs(double a) {
	if (a < 0.0) {
		return (-a);
	}
	else {
		return (a);
	}
}

public static float abs(float a) {
	if (a < 0.0f) {
		return (-a);
	}
	else {
		return (a);
	}
}

public static int abs(int a) {
	if (a < 0) {
		return (-a);
	}
	else {
		return (a);
	}
}

public static long abs(long a) {
	if (a < 0L) {
		return (-a);
	}
	else {
		return (a);
	}
}

native public static double acos(double a);

native public static double asin(double a);

native public static double atan(double a);

native public static double atan2(double a, double b);

native public static double ceil(double a);

native public static double cos(double a);

native public static double exp(double a);

native public static double floor(double a);

native public static double log(double a);

public static double max(double a, double b) {
	if (b > a) {
		return (b);
	}
	else {
		return (a);
	}
}

public static float max(float a, float b) {
	if (b > a) {
		return (b);
	}
	else {
		return (a);
	}
}

public static int max(int a, int b) {
	if (b > a) {
		return (b);
	}
	else {
		return (a);
	}
}

public static long max(long a, long b) {
	if (b > a) {
		return (b);
	}
	else {
		return (a);
	}
}

public static double min(double a, double b) {
	if (a < b) {
		return (a);
	}
	else {
		return (b);
	}
}

public static float min(float a, float b) {
	if (a < b) {
		return (a);
	}
	else {
		return (b);
	}
}

public static int min(int a, int b) {
	if (a < b) {
		return (a);
	}
	else {
		return (b);
	}
}

public static long min(long a, long b) {
	if (a < b) {
		return (a);
	}
	else {
		return (b);
	}
}

native public static double pow(double a, double b);

public static synchronized double random() {
	double dbl=randomGenerator.nextDouble();
	return dbl;
}

native public static double rint(double a);

public static long round(double a) {
	if ((a<(double )Long.MIN_VALUE) || (a==Double.NEGATIVE_INFINITY)) {
		return Long.MIN_VALUE;
	}
	else if ((a>(double )Long.MAX_VALUE) || (a==Double.POSITIVE_INFINITY)) {
		return Long.MAX_VALUE;
	}
	else {
		return (long )rint(a);
	}
}

public static int round(float a) {
	if ((a<(float )Integer.MIN_VALUE) || (a==Float.NEGATIVE_INFINITY)) {
		return Integer.MIN_VALUE;
	}
	else if ((a>(float )Integer.MAX_VALUE) || (a==Float.POSITIVE_INFINITY)) {
		return Integer.MAX_VALUE;
	}
	else {
		return (int )rint((double )a);
	}
}

native public static double sin(double a);

native public static double sqrt(double a);

native public static double tan(double a);
}
